frontend/public/module/k8s/cluster-settings.ts

```diff
@@ -256,3 +258,35 @@ export const getConditionUpgradeableFalse = (resource) =>
 
 export const getNotUpgradeableResources = (resources) =>
   resources.filter((resource) => getConditionUpgradeableFalse(resource));
+
+export enum NodeTypes {
+  master = 'master',
+  worker = 'worker',
+}
+
+export enum NodeTypeNames {
+  Master = 'Master',
+  Worker = 'Worker',
+}
+
+export const isMCPMaster = (mcp: MachineConfigPoolKind) => mcp.metadata.name === NodeTypes.master;
+
+export const isMCPWorker = (mcp: MachineConfigPoolKind) => mcp.metadata.name === NodeTypes.worker;
+
+export const isMCPPaused = (mcp: MachineConfigPoolKind) => mcp.spec.paused;
+
+export const sortMCPsByCreationTimestamp = (a: MachineConfigPoolKind, b: MachineConfigPoolKind) =>
+  a.metadata.creationTimestamp.localeCompare(b.metadata.creationTimestamp);
+
+export const clusterIsUpToDateOrUpdateAvailable = (status: ClusterUpdateStatus) =>
+  status === ClusterUpdateStatus.UpToDate || status === ClusterUpdateStatus.UpdatesAvailable;
+
+export const getMCPsToPausePromises = (
+  promises: Promise<any>[],
+  machineConfigPools: MachineConfigPoolKind[],
+  paused: boolean,
+) =>
+  machineConfigPools.map((mcp) => {
+    const patch = [{ op: 'add', path: '/spec/paused', value: paused }];
+    promises.push(k8sPatch(MachineConfigPoolModel, mcp, patch));
+  });
```

Comment:
Two notes here:

1. You shouldn't really mutate properties that are passed to you -- you should always create your own and return the value; this is because it's harder to track mutations when reading code than it is to read joining of values together.
2. `.map` always expects a value... not returning means you probably want a `.forEach`; JavaScript allows no returns, but our TypeScript/Linter (outside of public folder) will have a harder time with it (or it should for code quality)

Reply:
> You shouldn't really mutate properties that are passed to you

In this case, `promises`?

> you probably want a .forEach

Indeed.  Will change.

Reply:
That is correct, `promises` is mutated when you call `.push`. So the caller passes you an array and you mutate it. It's anti-pattern for functional programming to mutate a calling variable. In this case, you're doing both sides of the fence so you "know" that's the whole point of passing the array. But as code mutates over time, it gets less and less clear and the code is likely not to be easily reused.

`.forEach` is for side-effects... in this case your mutation of promises. But the most important is to avoid `.map` that never returns... because JS will return undefined... and then your method now returns an array of undefined equal to the length of `machineConfigPools`. Which is definitely not what you want to do.